example/src/main/java/com/stripe/example/activity/ConfirmSepaDebitActivity.kt

```diff
@@ -1,155 +1,65 @@
 package com.stripe.example.activity
 
-import android.content.Intent
 import android.os.Bundle
 import android.view.View
-import androidx.appcompat.app.AppCompatActivity
-import com.stripe.android.ApiResultCallback
+import androidx.lifecycle.Observer
 import com.stripe.android.PaymentIntentResult
-import com.stripe.android.Stripe
 import com.stripe.android.model.Address
-import com.stripe.android.model.ConfirmPaymentIntentParams
 import com.stripe.android.model.MandateDataParams
 import com.stripe.android.model.PaymentMethod
 import com.stripe.android.model.PaymentMethodCreateParams
 import com.stripe.example.R
-import com.stripe.example.StripeFactory
 import com.stripe.example.databinding.CreateSepaDebitActivityBinding
-import com.stripe.example.module.BackendApiFactory
-import com.stripe.example.service.BackendApi
-import io.reactivex.android.schedulers.AndroidSchedulers
-import io.reactivex.disposables.CompositeDisposable
-import io.reactivex.schedulers.Schedulers
-import org.json.JSONObject
 
 /**
  * An example integration for confirming a Payment Intent using a SEPA Debit Payment Method.
  *
  * See [SEPA Direct Debit payments](https://stripe.com/docs/payments/sepa-debit) for more
  * details.
  */
-class ConfirmSepaDebitActivity : AppCompatActivity() {
+class ConfirmSepaDebitActivity : StripeIntentActivity() {
     private val viewBinding: CreateSepaDebitActivityBinding by lazy {
         CreateSepaDebitActivityBinding.inflate(layoutInflater)
     }
 
-    private val compositeSubscription = CompositeDisposable()
-
-    private val backendApi: BackendApi by lazy {
-        BackendApiFactory(this).create()
-    }
-
     private val snackbarController: SnackbarController by lazy {
         SnackbarController(viewBinding.coordinator)
     }
 
-    private val stripe: Stripe by lazy {
-        StripeFactory(this).create()
-    }
-
-    private var clientSecret: String? = null
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(viewBinding.root)
         setTitle(R.string.launch_confirm_pm_sepa_debit)
 
+        viewModel.inProgress.observe(this, Observer { enableUi(!it) })
+        viewModel.status.observe(this, Observer(viewBinding.status::setText))
+
         viewBinding.ibanInput.setText(TEST_ACCOUNT_NUMBER)
 
         viewBinding.confirmButton.setOnClickListener {
-            viewBinding.status.text = ""
-            createPaymentIntent(viewBinding.ibanInput.text.toString())
+            val params =
+                createPaymentMethodParams(viewBinding.ibanInput.text.toString())
+                    .takeIf { EXISTING_PAYMENT_METHOD_ID == null }
+            createAndConfirmPaymentIntent("nl", params,
```

Comment:
optional nit: I've been putting one formatting like this:

```
            createAndConfirmPaymentIntent(
                "nl",
                params,
                existingPaymentMethodId = EXISTING_PAYMENT_METHOD_ID,
                mandateDataParams = mandateData
            )
```

probably good time to define this in a style guide :)